DESIGN.md

```diff
@@ -86,10 +86,88 @@ requesting a release and assigning the bot.
 
 The `shipit init` command initializes a project for Shipit. The project must
 already have a git repository initialized with Github as the origin. The output
-is a `.shipit.yml` file at the root of the root of the repository.
+is a `.shipit.yml` file at the root of the root of the repository with project
+specific configuration values.
 
+If initializing a crate that already has published releases to crates.io, then
+.`shipit.yml` will include information indicating at what point in the history
+Shipit was added. This is the first unpublished crate version and an optional
+git sha. This information helps Shipit integrate with existing projects.
 
+## Pull requests
 
+### Opening a new pull request
+
+When a new pull request is opened, the Shipit bot is notified via a Github
+status check notification. The Github bot verifies the PR has the appropriate
+labels. If the labels are not correctly set, the pull request cannot be merged
+as the status check will fail.
+
+If the pull request includes the `breaking-change` label, then the status check
+ensures that the crate version has been incremented as part of the PR to reflect
+the breaking change.
+
+### Merging a pull request
+
+When merging a pull request, the merger should ensure that the title of the pull
+request is appropriate for a changelog entry. When publishing, Shipit will use
+this title for the entry.
+
+### Coordinating many breaking changes
+
+Most projects batch breaking changes into a single breaking release. To do this,
+breaking changes should be staged on a branch. Pull requests **must** be used to
+apply changes to this branch in the same way as described above. When the
+project is ready to publish the breaking changes, a pull request is opened to
+merge the long lived branch into the master branch.
+
+**When merging the pull request, a merge commit should be used instead of
+rebasing**
+
+## Publishing changes
+
+Describes the automated process of releasing crates
+
+### Opening a release pull request
+
+At a regular interval, the Shipit bot will open pull requests to publish crates.
+This pull request will make the following changes:
+
+* Increment the crate version (if needed)
+* Update the documentation URL if docs.rs is used.
+* Generate changelog entries.
+
+In the git commit, Shipit leaves the following message:
+
+```
+Release [crate-name] vX.X.X
+
+Shipit-release: [crate-name]-X.X.X
+```
+
+If multiple crates in a workspace have unpublished changes, a single commit is
+created for all of them. In this case, the message will be:
+
+```
+Release [crate-a], [crate-b], [crate-c]
+
+Shipit-release: [crate-a]-X.X.X
+Shipit-release: [crate-b]-X.X.X
+Shipit-release: [crate-c]-X.X.X
+```
+
+The `Shipit-release` tag allows Shipit to identify commits that it has authored
+for the purpose of releasing crates.
+
+When the project maintainers are ready to publish, the pull request is merged.
+
+### Publish process
+
+TODO: what happens on CI to publish the crates.
```

Comment:
One thing, that would be good is when CI publishes the changes it should also push a tag for each crate and its version to github.